### PR TITLE
Add Dynamic Allocation on YARN notes.

### PR DESCRIPTION
**Pull request by eliasah**

### CODE REVIEW

**Review by zero323 — CHANGES_REQUESTED** [review]
I believe it should go to the Spark Application Deployment. There is nothing that justifies a seperate chapter.

File: 11_small_deployments_and_dynamic_allocation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Spark offers yarn specific properties so you can run your application :
+
+- `spark.yarn.executor.memoryOverhead` is the amount of off-heap memory (in megabytes) to be allocated per executor. This is memory that accounts for things like VM overheads, interned strings, other native overheads, etc. This tends to grow with the executor size (typically 6-10%).
+- `spark.yarn.driver.memoryOverhead` is the amount of off-heap memory (in megabytes) to be allocated per driver in cluster mode with the memory properties as the executor's memoryOverhead.
+
+So it's not about storing data, it's just the resources needed for YARN to run properly.
+
+In some cases,
+
+----------
```
Comment (zero323) [inline 11_small_deployments_and_dynamic_allocation.md]: Can we avoid artificial formatting?

File: 11_small_deployments_and_dynamic_allocation.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Small deployments and dynamic allocation [WIP]
+
+## Dynamic Allocation
+
+### Yet Another Dynamic Resource Allocation YA(D)RN
```
Comment (zero323) [inline 11_small_deployments_and_dynamic_allocation.md]: Do we need a sub section here? Do you plan to add more content?

File: 11_small_deployments_and_dynamic_allocation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Small deployments and dynamic allocation [WIP]
```
Comment (zero323) [inline 11_small_deployments_and_dynamic_allocation.md]: WIP  makes sense for PR title not here. All of that is WIP

File: 11_small_deployments_and_dynamic_allocation.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+- `spark.dynamicAllocation.maxExecutors` is set to infinity by default which set the upper bound for the number of executors if dynamic allocation is enabled. Ref. [Official documentation - Configuration](http://spark.apache.org/docs/latest/configuration.html#dynamic-allocation).
+
+- According to the code documentation : Ref. [ExecutorAllocationManager.scala](https://github.com/apache/spark/blob/8ef3399aff04bf8b7ab294c0f55bcf195995842b/core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala#L43).
+
+Increasing the target number of executors happens in response to backlogged tasks waiting to be scheduled. If the scheduler queue is not drained in N seconds, then new executors are added. If the queue persists for another M seconds, then more executors are added and so on. The number added in each round increases exponentially from the previous round until an upper bound has been reached. The upper bound is based both on a configured property and on the current number of running and pending tasks, as described above.
+
+This can lead into an exponential increase of the number of executors in some cases which can break the YARN resource manager. In my case :
```
Comment (zero323) [inline 11_small_deployments_and_dynamic_allocation.md:21]: I would prefer more authoritative style than personal.

File: 11_small_deployments_and_dynamic_allocation.md (hunk 2 above)
Comment (eliasah) [reply 11_small_deployments_and_dynamic_allocation.md]: I was thinking about a section for Mesos if it's possible.

File: 11_small_deployments_and_dynamic_allocation.md (hunk 3 above)
Comment (eliasah) [reply 11_small_deployments_and_dynamic_allocation.md]: I thought that it was needed for the toc generator

File: 11_small_deployments_and_dynamic_allocation.md (hunk 1 above)
Comment (eliasah) [reply 11_small_deployments_and_dynamic_allocation.md]: What's artificial formatting ?

File: 11_small_deployments_and_dynamic_allocation.md (hunk 4 above)
Comment (eliasah) [reply 11_small_deployments_and_dynamic_allocation.md:21]: I'll review this paragraph.

File: 11_small_deployments_and_dynamic_allocation.md (hunk 2 above)
Comment (zero323) [reply 11_small_deployments_and_dynamic_allocation.md]: Wouldn't make more sense h2 for YARN, h2 for Mesos?

File: 11_small_deployments_and_dynamic_allocation.md (hunk 1 above)
Comment (zero323) [reply 11_small_deployments_and_dynamic_allocation.md]: Horizontal line you include with `---------`

File: 11_small_deployments_and_dynamic_allocation.md (hunk 4 above)
Comment (zero323) [reply 11_small_deployments_and_dynamic_allocation.md:21]: I think we should the same in MySQL section. For example "@zero323 comment on that : When you create data frame it fetches schema from database, and the column used has most likely NOT NULL constraint." sounds like a Tweet :)

Comment (eliasah) [reply 11_small_deployments_and_dynamic_allocation.md:21]: I don't get your last comment.

**Comment by zero323** [comment]
A thought - it would be nice to explain how dynamic allocation affects static objects and PySpark resources.

File: 11_small_deployments_and_dynamic_allocation.md (hunk 4 above)
Comment (eliasah) [reply 11_small_deployments_and_dynamic_allocation.md:21]: Just a little reminder about this. You want me to rephrase that sentence ?

**Comment by zero323** [comment]
Does it mean you consider this to be ready for merge?

**Comment by eliasah** [comment]
I'm not sure that's why I'm asking you. I'm not sure that I know what static objects are with python nor if you want me to rephrase the sentence you cited.

**Comment by zero323** [comment]
OK.

**Comment by eliasah** [comment]
Ok what ? You are confusing me. :)

**Comment by zero323** [comment]
I acknowledged your statement :)

**Comment by eliasah** [comment]
Ok then I'm merging

**Pull request merged**